### PR TITLE
feat(glue-alpha): add job run queuing to Glue job

### LINKED ISSUE #31826: (aws-glue-alpha): (add jobRunQueuingEnabled to Job)
### Describe the feature

Add the [jobRunQueuingEnabled](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue.CfnJob.html#jobrunqueuingenabled) property of [CfnJob](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue.CfnJob.html) within [aws-cdk-lib/aws-glue](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue-readme.html) to [Job](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-glue-alpha.Job.html) within [@aws-cdk/aws-glue-alpha](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-glue-alpha-readme.html).

### Use Case

Setting this property currently requires a [raw override](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html#develop-customize-override); for example

```ts
(job.node.defaultChild as CfnJob).addOverride('Properties.JobRunQueuingEnabled', true)
```

### Proposed Solution

_No response_

### Other Information

_No response_

### Acknowledgements

- [X] I may be able to implement this feature request
- [X] This feature might incur a breaking change

### CDK version used

v2.162.1

### Environment details (OS name and version, etc.)

macOS Sequoia 15.0.1

### PR DESCRIPTION
**Pull request by justindebuse**
### Issue # (if applicable)

Closes #31826 

### Reason for this change

[Job](https://docs.aws.amazon.com/cdk/api/v2/docs/@aws-cdk_aws-glue-alpha.Job.html) within [@aws-cdk/aws-glue-alpha](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-glue-alpha-readme.html) does not currently include the [jobRunQueuingEnabled](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue.CfnJob.html#jobrunqueuingenabled) property of the [CfnJob](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue.CfnJob.html) within [aws-cdk-lib/aws-glue](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_glue-readme.html). Setting this property currently requires a [raw override](https://docs.aws.amazon.com/cdk/v2/guide/cfn_layer.html#develop-customize-override).

### Description of changes

Added `jobRunQueuingEnabled` to construction properties for `Job`, along with validation that this is not enabled when execution class is flexible and/or `maxRetries` exceeds zero  ([see](https://aws.amazon.com/blogs/big-data/introducing-job-queuing-to-scale-your-aws-glue-workloads/)).

### Description of how you validated changes

Unit tests and an integration test.

### Checklist
- [x] My code adheres to the [CONTRIBUTING GUIDE](https://github.com/aws/aws-cdk/blob/main/CONTRIBUTING.md) and [DESIGN GUIDELINES](https://github.com/aws/aws-cdk/blob/main/docs/DESIGN_GUIDELINES.md)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by jdebuseamazon** [comment]
Exemption Request (for `❌ Features must contain a change to a README file`): the change only introduced an optional construction property, and for consistency with existing similar construction properties such as `maxCapacity` and `maxRetries` no details were included in the README file

**Comment by badmintoncryer** [comment]
@jdebuseamazon Thank you for your contribution! I think it would be helpful to add a README description for this feature.

The descriptions for maxCapacity and minCapacity are necessary, but this should not be addressed in this PR.

File: packages/@aws-cdk/aws-glue-alpha/test/job.test.ts
Comment (badmintoncryer) [inline packages/@aws-cdk/aws-glue-alpha/test/job.test.ts:824]: This description may be sufficient.

```suggestion
    test('enabling job run queuing', () => {
        job = new glue.Job(stack, 'Job', {
          ...defaultProps,
          jobRunQueuingEnabled: true,
        });
        Template.fromStack(stack).hasResourceProperties('AWS::Glue::Job', {
          JobRunQueuingEnabled: true,
        });
    });
```

File: packages/@aws-cdk/aws-glue-alpha/test/job.test.ts
Comment (badmintoncryer) [inline packages/@aws-cdk/aws-glue-alpha/test/job.test.ts:1208]: If there is only one test within a describe block, I think a single test block is sufficient.

File: packages/@aws-cdk/aws-glue-alpha/lib/job.ts
Comment (badmintoncryer) [inline packages/@aws-cdk/aws-glue-alpha/lib/job.ts:759]: Please add condition `!Token.isUnresolved(props.maxRetries)`  to the if statement. 

This will enable support when `maxRetries` is token.

```suggestion
    if (props.jobRunQueuingEnabled === true && props.maxRetries !== undefined && Token.isUnresolved(props.maxRetries) && props.maxRetries > 0) {
```

**Review by badmintoncryer — CHANGES_REQUESTED** [review]
Thank you for your contribution!

I would appreciate it if you could provide links regarding the need for a max retry of 0 and the fact that queueing cannot be enabled with the FLEX ExecutionClass.

**Comment by jdebuseamazon** [comment]
> Thank you for your contribution!
> 
> I would appreciate it if you could provide links regarding the need for a max retry of 0 and the fact that queueing cannot be enabled with the FLEX ExecutionClass.

No problem at all, very happy to contribute, and many thanks for reviewing. The restrictions on max retry and FLEX execution class are documented [here](https://aws.amazon.com/blogs/big-data/introducing-job-queuing-to-scale-your-aws-glue-workloads/) (in the _Considerations_ section)

**Review by badmintoncryer — APPROVED** [review]
Thank you for the fix! The documentation you provided was also very helpful.

By merging updated base branch as needed, the "needs-maintainer-review" label will be applied. After that, please wait for the maintainer's review😁

**Comment by paulhcsun** [comment]
@mergifyio update

**Pull request merged**
